source/crud/crud.rst

```diff
@@ -1953,54 +1953,37 @@ in the collection prior to ``$merge`` being executed).
 
+This section is only applicable if an explicit (i.e. per-operation) or inherited
+(e.g. from a Collection) read preference is available and it is *not* a primary
+read preference (i.e. ``{ "mode": "primary" }``).
+
 Historically, only primaries could execute an aggregation pipeline with ``$out``
-or ``$merge``. As of ``featureCompatibilityVersion`` 4.4, secondaries can now
-execute such pipelines; however, since drivers do not track
-``featureCompatibilityVersion``, executing an aggregation pipeline with ``$out``
-or ``$merge`` on a secondary will require a 5.0+ server (i.e. wire version >=
-13).
-
-Drivers MUST guarantee the following:
-
-- If an explicit (i.e. per-operation) read preference is specified for an
-  aggregation with a write stage, drivers MUST attempt to use it. If that would
-  result in a pre-5.0, secondary server being selected, drivers MUST instead
-  select a server using a primary read preference.
-- If no explicit read preference is specified but a default read preference is
-  available to inherit (e.g. from the Collection), drivers MUST attempt to use
-  it. If that would result in a pre-5.0, secondary server being selected,
-  drivers MUST instead select a server using a primary read preference.
-
-Drivers SHOULD use a custom server selector to consider server/wire version when
-matching a read preference and, if a pre-5.0 secondary would be selected, fall
-back to selecting a primary. With this approach, only a single server selection
-attempt is needed.
-
-If it is not possible to augment the server selector and a pre-5.0 secondary is
-selected, drivers MAY invoke server selection a second time to obtain a primary.
-However, if the first attempt results in a server selection timeout, drivers
-MUST NOT make a second attempt and MUST propagate the original timeout error. If
-the topology type is Single, Sharded, or LoadBalanced, drivers MUST NOT make a
-second attempt (there is no benefit in doing so).
-
-Regardless of whether drivers employ a custom server selector or two-attempt
-approach, they MUST discern the *effective* read preference for the operation,
-which SHALL be used for specifying the
+or ``$merge`` and drivers never considered a read preference for the operation.
+As of ``featureCompatibilityVersion`` 4.4, secondaries can now execute pipelines
+with ``$out`` or ``$merge``. Since drivers do not track
+``featureCompatibilityVersion``, the decision to consider a read preference for
+such a pipeline will depend on the wire version(s) of the server(s) to which the
+driver is connected.
+
+If there are one or more available servers and one or more of those servers is
+pre-5.0 (i.e. wire version < 13), drivers MUST NOT use the available read
+preference and MUST instead select a server using a primary read preference.
+Otherwise, if there are either no available servers or all available servers are
+5.0+ (i.e. wire version >= 13), drivers MUST use the available read preference.
```

Comment:
@rstam originally proposed:

> If there are any available servers and all available servers are 5.0+ apply the read preference; otherwise, fallback to primary

This is the text that @jyemin and I devised after discussing over Zoom. I don't believe there's any functional difference, as the edge case if when there are no available servers with which to make a decision and in that case the user is going to encounter a server selection failure regardless.

In contrast to the original text, these new rules do mean that there is now no change that a capable 5.0+ secondary might be selected while the driver is connected to a mixed version replica set. I personally didn't see the issue with the previous behavior but I won't object if this is what others are asking for.

The only concern I had with the original text was the BC issue when the driver was connected to a single pre-5.0 primary. With the original text, the user would always encounter a server selection failure if the operation was provided RP(secondary).